### PR TITLE
MVP-156: Error Wrapper

### PR DESCRIPTION
**Pull request by BradyMitch**
## 🎯 Summary

  
[MVP-156](https://citz-imb.atlassian.net/jira/browse/MVP-156)

- Added HttpError error instance class
- Added HTTP_STATUS_CODES enum
- Added errorWrapper function

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](/CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - I have consulted with the team if introducing a new dependency.
> - My changes generate no new warnings.

### CODE REVIEW

File: src/constants/errorWrapperOptions.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { RouteHandlerErrorProperties } from '../types';
+
+export const DEFAULT_CUSTOM_LOG_FUNCTION = ({
+  method,
+  originalUrl,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
+  statusCode,
```
Comment (SDToews) [inline src/constants/errorWrapperOptions.ts]: You don't need to destructure statusCode 
```suggestion
```

File: src/constants/errorWrapperOptions.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  message,
+}: RouteHandlerErrorProperties) => {
+  console.error(`REQUEST ERROR: [${method}] ${originalUrl}: ${message}`);
+};
+export const DEFAULT_CUSTOM_JSON_RESPONSE = ({
+  method,
+  originalUrl,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
+  statusCode,
```
Comment (SDToews) [inline src/constants/errorWrapperOptions.ts]: You don't need to destructure statusCode 

```suggestion
```

File: src/types.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  res: Response,
+  next: NextFunction,
+) => Promise<void | Response<unknown> | undefined>;
+
+export type HttpStatusCode = (typeof HTTP_STATUS_CODES)[keyof typeof HTTP_STATUS_CODES];
+
+export type RouteHandlerErrorProperties = {
+  method: string;
+  originalUrl: string;
+  statusCode: HttpStatusCode | number;
```
Comment (SDToews) [inline src/types.ts:16]: might want to consider making this optional
```suggestion
  statusCode?: HttpStatusCode | number;
```

**Review by SDToews — CHANGES_REQUESTED** [review]

File: src/constants/errorWrapperOptions.ts (hunk 1 above)
Comment (BradyMitch) [reply src/constants/errorWrapperOptions.ts]: I'm making it available to the custom function, just not using it in the default.

File: src/constants/errorWrapperOptions.ts (hunk 2 above)
Comment (BradyMitch) [reply src/constants/errorWrapperOptions.ts]: Same as above.

File: src/constants/errorWrapperOptions.ts (hunk 1 above)
Comment (BradyMitch) [reply src/constants/errorWrapperOptions.ts]: oh nvm I see why it can be omitted here

**Review by SDToews — APPROVED** [review]

**Pull request merged**